### PR TITLE
Beginner tutorial

### PR DESCRIPTION
**Pull request by codeliner**
This PR adds the first page of the new walk-through guide. Additionally I've added new composer commands and switched the bookdown template to use our fork.

I also played with the available bootswatch templates and like the `Ỳeti` template but this is subject to discuss ;)

Everything of this PR can be discussed. I just wanted to start somewhere so I've created a develop branch where we can merge things until we have a final structure for getprooph.org

Please read the first part of the tutorial, correct my English when needed and give feedback if contains too much or less details and so on and so on...

### CODE REVIEW

**Comment by mablae** [comment]
Nice start, very well written!

File: composer.json [hunk 1]
```diff
@@ -22,6 +22,11 @@
         "prooph"
     ],
     "require-dev": {
-        "tobiju/bookdown-bootswatch-templates": "^0.2.0"
+        "prooph/bookdown-template": "^0.2.0",
+        "bookdown/bookdown": "1.0.0-beta1 as 1.0.0"
+    },
+    "scripts": {
+        "build": "export CSS_BOOTSWATCH=yeti && export CSS_PRISM=coy && vendor/bin/bookdown docs/bookdown.json",
+        "serve": "php -S 0.0.0.0:8080 -t docs/html/"
```
Comment (sandrokeil) [inline composer.json]: We should replace `build` with `docs`.

File: docs/book/tutorial/getting-started/beginner_tutorial.md [hunk 2]
```diff
@@ -0,0 +1,492 @@
...
+In the prooph ecosystem everything is bound together by messages.
+So when you want to get started with prooph components you should know the
+basic building block - **prooph messages**. Ones you know how they work we give a short overview
+of CQRS and Event Sourcing and why messages play an important role in this architecture.
+
+## The initial setup
+
+To follow the tutorial you need PHP 7.1 and composer installed. Later you'll also need some more
+infrastructure like a Postgres database server. Nowadays Docker makes it easy to run infrastructure on demand.
+While Docker is not required for prooph components we recommend using it. All examples in the tutorial
```
Comment (sandrokeil) [inline docs/book/tutorial/getting-started/beginner_tutorial.md]: Add link to Docker website for `Nowadays [Docker](https://www.docker.com/)`

**Comment by sandrokeil** [comment]
@codeliner You can also use tables and attributes if you use these [Common Mark extensions](https://github.com/bookdown/Bookdown.Bookdown/blob/1.x/tests/Config/RootConfigTest.php#L23-L28).

File: docs/book/tutorial/getting-started/beginner_tutorial.md [hunk 3]
```diff
@@ -0,0 +1,492 @@
...
+        }
+        
+        return $user;
+    }
+}
+
+```
+
+The `ÙserService` is responsible for all actions related to a user. 
+In a CQRS system this looks slightly different:
```
Comment (sandrokeil) [inline docs/book/tutorial/getting-started/beginner_tutorial.md]: Typo in `ÙserService`, replace `Ù` with `U`.

File: docs/book/tutorial/getting-started/beginner_tutorial.md [hunk 4]
```diff
@@ -0,0 +1,492 @@
...
+ 
+We no longer use the `UserRepository` but instead directly access the database.
+So we avoid the object relational mapping and return pure user data from our finder.
+We can do this because we know that our read model won't do anything with the data other than
+forwarding it to a client that for example requires the data in JSON or XML format.
+If it is quaranteed that no state changes happen within the read model, we don't need to deal with 
+objects as we don't need to enforce any rules. Select the data and return it to the client as fast as possible
+that is the target of the read model.
+
+The write model however has to protect invariants. At the moment our `Ùser` object does a bad job on this.
```
Comment (sandrokeil) [inline docs/book/tutorial/getting-started/beginner_tutorial.md]: Typo in `Ùser`, replace `Ù` with `U`

File: docs/book/tutorial/getting-started/beginner_tutorial.md [hunk 5]
```diff
@@ -0,0 +1,492 @@
...
+    {
+        $user = User::register($command->userId(), $command->userName(), $command->email());
+
+        $this->userRepository->save($user);
+    }
+}
+
+```
+
+With a few changes we turned our original `ÙserService` into two distinct classes supporting the basic idea of CQRS.
```
Comment (sandrokeil) [inline docs/book/tutorial/getting-started/beginner_tutorial.md]: Typo in `ÙserService`, replace `Ù` with `U`

File: docs/book/tutorial/getting-started/beginner_tutorial.md [hunk 6]
```diff
@@ -0,0 +1,492 @@
...
+In the last step we enabled the write side to handle a prooph command that expresses its intent of how the system state should change
+(a new user should be registered) using the message name `RegisterUser` and payload of the command.
+The `RegisterUserHandler` is a so called *glue component*. Its task is to take the command and
```
Comment (sandrokeil) [inline docs/book/tutorial/getting-started/beginner_tutorial.md]: Typo in `Its`, must be `It's`

File: docs/book/tutorial/getting-started/beginner_tutorial.md [hunk 7]
```diff
@@ -0,0 +1,492 @@
...
+
+```
+id  | name     | email
+--- | -------- | ------------
+1   | John Doe | doe@test.com
+```
+
+
+Applying CQRS again we end up with a new command `ChangeEmail`, an appropriate command handler and a matching action
+in the write model owned by the responsible object `Ùser::changeEmail`
```
Comment (sandrokeil) [inline docs/book/tutorial/getting-started/beginner_tutorial.md]: Simply replace all occurences of `Ù` with `U` ;-)

**Comment by sandrokeil** [comment]
Awesome :+1:

File: composer.json [hunk 8]
```diff
@@ -22,6 +22,11 @@
         "prooph"
     ],
     "require-dev": {
-        "tobiju/bookdown-bootswatch-templates": "^0.2.0"
+        "prooph/bookdown-template": "^0.2.0",
```
Comment (prolic) [inline composer.json]: v0.2.3 - sonst schlechtes design

File: docs/book/tutorial/bookdown.json [hunk 9]
```diff
@@ -0,0 +1,6 @@
+{
+  "title": "Tutorial",
+  "content": [
+    {"getting-started": "getting-started/bookdown.json"}
```
Comment (prolic) [inline docs/book/tutorial/bookdown.json]: Where to link to component docs?

File: docs/book/tutorial/getting-started/beginner_tutorial.md [hunk 10]
```diff
@@ -0,0 +1,492 @@
...
+### The first message
+
+Create a file called `hello_world.php` with the following content and run it with `php`:
+
+```php
+<?php
+//All prooph components use strict types enabled
+declare(strict_types=1);
+
+namespace ProophTutorial;
```
Comment (prolic) [inline docs/book/tutorial/getting-started/beginner_tutorial.md]: Prooph\Tutorial maybe?

File: docs/book/tutorial/getting-started/beginner_tutorial.md [hunk 11]
```diff
@@ -0,0 +1,492 @@
...
+     * Returns new instance of message with $key => $value added to metadata
+     *
+     * Given value must have a scalar or array type.
+     */
+    public function withAddedMetadata(string $key, $value): Message;
+}
+```
+That is the basic contract, an immutable message with a unique identifier, a type, an created at timestamp, 
+a message name (by extending the `HasMessageName` interface), payload and metadata.
+Both payload and metadata should only contain scalar types and arrays (no objects) to be truly immutable.
```
Comment (prolic) [inline docs/book/tutorial/getting-started/beginner_tutorial.md]: Note: The abstract class `DomainMessage` does only allow scalar values for metadata, not arrays!

**Comment by codeliner** [comment]
thx for all the feedback and the good hints. Will incorporate them soon.

File: docs/book/tutorial/getting-started/beginner_tutorial.md [hunk 12]
```diff
@@ -0,0 +1,492 @@
...
+On the following pages we will walk through all prooph components step by step 
+and take a look on how things work together. You will learn the basic concepts as well
+as some advanced techniques to push your PHP applications up to the next level and 
+make them ready for the future.
+
+## Objectives of this tutorial
+
+In the prooph ecosystem everything is bound together by messages.
+So when you want to get started with prooph components you should know the
+basic building block - **prooph messages**. Ones you know how they work we give a short overview
```
Comment (danizord) [inline docs/book/tutorial/getting-started/beginner_tutorial.md]: s/Ones/Once

File: docs/book/tutorial/getting-started/beginner_tutorial.md [hunk 13]
```diff
@@ -0,0 +1,492 @@
...
+Finally, the command handler makes use of the infrastructure (represented by the `UserRepository`) to persist 
+the state change caused by the command.
+
+## Event Sourcing
+
+Event Sourcing is covered in detail in the next chapter. What follows is a very basic introduction so that
+you can get an idea of what you'll learn when continue reading.
+
+In an event sourced system all state changes are described by events. The fact that a new user was registered in
+our system would be one of those events another one would be that the user has logged in or changed the email address.
```
Comment (danizord) [inline docs/book/tutorial/getting-started/beginner_tutorial.md]: "another" should start a new sentence

**Comment by codeliner** [comment]
@sandrokeil any ideas how I can add classes to a table? I've included the extensions and a standard html table is rendered but without bootstrap classes so it looks not really nice.

**Comment by codeliner** [comment]
@sandrokeil forget my question. just need to combine table and attribute extension ;)
```
{.table}
id  | name     | email
--- | -------- | ------------
1   | John Doe | doe@test.com
```

File: docs/book/tutorial/bookdown.json (hunk 9 above)
Comment (codeliner) [inline docs/book/tutorial/bookdown.json]: in  docs/book/components/bookdown.json

File: docs/book/tutorial/getting-started/beginner_tutorial.md (hunk 6 above)
Comment (codeliner) [inline docs/book/tutorial/getting-started/beginner_tutorial.md]: hhhm, I think `Its` is correct here. In German: "Seine Aufgabe ist es ..."

**Comment by codeliner** [comment]
Ok, should be ready now. Thx all reviewers!

**Comment by sandrokeil** [comment]
@codeliner FYI: [PR to disable numbering](https://github.com/bookdown/Bookdown.Bookdown/pull/56)

**Pull request merged**